commit 82fc75a3421455fb4a3542fd6bf790f703741d23 (refs/changes/23/3923/1)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2018-05-09T12:03:59-04:00 (1 year, 5 months ago)
    
    OS-6942 bhyve incapable hosts should exit early on vmadm create

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index d82ab6d8..416316d5 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -10036,6 +10036,35 @@ exports.create = function (payload, options, callback)
         }, function (cb) {
             var vm_type = BRAND_OPTIONS[payload.brand].features.type;
 
+            switch (vm_type) {
+            case 'BHYVE':
+                VM.getSysinfo(function gotSysinfo(err, sysinfo) {
+                    if (err) {
+                        cb(err);
+                        return;
+                    }
+
+                    if (!sysinfo.hasOwnProperty('Bhyve Capable')) {
+                        cb(new Error(
+                            'sysinfo does not have "Bhyve Capable" property'));
+                        return;
+                    }
+
+                    if (!sysinfo['Bhyve Capable']) {
+                        cb(new Error('Bhyve not supported'));
+                        return;
+                    }
+
+                    cb();
+                });
+                break;
+            default:
+                cb();
+                break;
+            }
+        }, function (cb) {
+            var vm_type = BRAND_OPTIONS[payload.brand].features.type;
+
             if (['BHYVE', 'KVM'].indexOf(vm_type) !== -1) {
                 createVM(payload, log, function (error, result) {
                     if (error) {
